mesonbuild/envconfig.py

```diff
@@ -400,9 +400,6 @@ def __init__(
         self.binaries: T.Dict[str, T.List[str]] = {}
         if binaries:
             for name, command in binaries.items():
-                if not isinstance(command, (list, str)):
-                    raise mesonlib.MesonException(
-                        f'Invalid type {command!r} for entry {name!r} in cross file')
```

Comment:
Drop this commit altogether. It isn't unused, no matter what the typing says. For one simple reason:

The typing says how this function is supposed to be used. It doesn't say that external users are *unable* to use it incorrectly. And this function's input is... a machine file specified on the command line. This code right here is the exclusive code for validating user input.

Reply:
Done